### PR TITLE
Added unittests for ServerContext

### PR DESCRIPTION
**Pull request by hod-alpert**
Adding unittests on the server context that will run a full test cycle.

- Implemented a demisto client mock
- Implemented a unit test that covers all test flow (except for mocks and locking mechanism).

This PR modifies only test files and requirement.txt file.

### CODE REVIEW

File: demisto_sdk/commands/test_content/tests/DemistoClientMock_test.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import re
```
Comment (Itay4) [inline demisto_sdk/commands/test_content/tests/DemistoClientMock_test.py:1]: dont think this file should have the `_test.py` suffix in its name

File: demisto_sdk/commands/test_content/tests/DemistoClientMock_test.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    INTEGRATION_SEARCH_PATTERN = re.compile(r'/settings/integration/(.*)')
+    INCIDENT_QUERY_PATTERN = re.compile(r'id: ([\d]+)')
+    GENERIC_RESPONSE = ("{}", 200, "")
+    INSTANCES = []
+    CONFIGURATIONS = []
+    api_client = MagicMock()
+    api_client.configuration.host = 'https://1.1.1.1'
+
+    @staticmethod
+    def create_incident(create_incident_request):
```
Comment (Itay4) [inline demisto_sdk/commands/test_content/tests/DemistoClientMock_test.py:21]: `create_incident_request` is not used

File: demisto_sdk/commands/test_content/tests/DemistoClientMock_test.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
+import re
+from unittest.mock import MagicMock
+
+from demisto_sdk.commands.common.constants import PB_Status
+from munch import Munch
```
Comment (Itay4) [inline demisto_sdk/commands/test_content/tests/DemistoClientMock_test.py:5]: its a bit unclear to me why we need both magicmock and munch

File: demisto_sdk/commands/test_content/tests/ServerContext_test.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import sys
```
Comment (Itay4) [inline demisto_sdk/commands/test_content/tests/ServerContext_test.py:1]: please keep filenames according to pep8 standard, for example `server_context_test.py`

File: demisto_sdk/commands/test_content/tests/ServerContext_test.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        - Running execute_tests.
+
+    Then:
+        - Ensure the server context has executed all the tests
+        - Ensure the mockable and unmockable tests queues were emptied
+        - Ensure no test has failed during that test
+    """
+    python_version = sys.version_info
+    if python_version.major == 3 and python_version.minor == 7:
+        pytest.skip('The mock required for this test is supported only in python 3.8+')
```
Comment (Itay4) [inline demisto_sdk/commands/test_content/tests/ServerContext_test.py:32]: why is that?

File: demisto_sdk/commands/test_content/tests/DemistoClientMock_test.py (hunk 1 above)
Comment (hod-alpert) [reply demisto_sdk/commands/test_content/tests/DemistoClientMock_test.py:1]: The precommit screamed at me for it.
Do you want me to try figuring out how to ignore it? or leave it as is?

File: demisto_sdk/commands/test_content/tests/ServerContext_test.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    integrations = [generate_integration_configuration('mockable_integration'),
+                    generate_integration_configuration('unmockable_integration')]
+    secret_test_conf = generate_secret_conf_json(integrations)
+    build_context = get_mocked_build_context(mocker,
+                                             tmp_path,
+                                             content_conf_json=content_conf_json,
+                                             secret_conf_json=secret_test_conf,
+                                             filtered_tests_content=filtered_tests)
+    # Setting up the client
+    mocker.patch('demisto_client.configure', return_value=DemistoClientMock)
```
Comment (Itay4) [inline demisto_sdk/commands/test_content/tests/ServerContext_test.py:54]: `DemistoClientMock` should mock demisto client end to end

File: demisto_sdk/commands/test_content/tests/ServerContext_test.py [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    mocker.patch('demisto_client.generic_request_func', side_effect=DemistoClientMock.generic_request_func)
+    DemistoClientMock.add_integration_configuration('mockable_integration')
```
Comment (Itay4) [inline demisto_sdk/commands/test_content/tests/ServerContext_test.py:56]: lets get rid of these lines, and initialize the mock client with the integrations the server should return

File: demisto_sdk/commands/test_content/tests/ServerContext_test.py [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    DemistoClientMock.add_integration_configuration('unmockable_integration')
+    # Mocking unnecessary calls
+    mocker.patch('demisto_sdk.commands.test_content.mock_server.run_with_mock').return_value.__enter__.return_value = {}
```
Comment (Itay4) [inline demisto_sdk/commands/test_content/tests/ServerContext_test.py:59]: lets add a func that will include all these mock

File: demisto_sdk/commands/test_content/tests/ServerContext_test.py [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+        - Ensure the server context has executed all the tests
+        - Ensure the mockable and unmockable tests queues were emptied
+        - Ensure no test has failed during that test
+    """
+    python_version = sys.version_info
+    if python_version.major == 3 and python_version.minor == 7:
+        pytest.skip('The mock required for this test is supported only in python 3.8+')
+
+    # Setting up the build context
+    filtered_tests = ['playbook_without_integrations',
```
Comment (Itay4) [inline demisto_sdk/commands/test_content/tests/ServerContext_test.py:35]: lets test skipped test

File: demisto_sdk/commands/test_content/tests/ServerContext_test.py [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    DemistoClientMock.add_integration_configuration('mockable_integration')
+    DemistoClientMock.add_integration_configuration('unmockable_integration')
+    # Mocking unnecessary calls
+    mocker.patch('demisto_sdk.commands.test_content.mock_server.run_with_mock').return_value.__enter__.return_value = {}
+    mocker.patch('demisto_sdk.commands.test_content.IntegrationsLock.safe_lock_integrations', return_value=True)
+    mocker.patch('demisto_sdk.commands.test_content.IntegrationsLock.safe_unlock_integrations')
+    mocker.patch('demisto_sdk.commands.test_content.TestContentClasses.TestContext._run_docker_threshold_test')
+    mocker.patch.object(MITMProxy, '__init__', lambda *args, **kwargs: None)
+    mocker.patch('time.sleep')
+
```
Comment (Itay4) [inline demisto_sdk/commands/test_content/tests/ServerContext_test.py:65]: lets see if we can create a file test to read from

**Review by Itay4 — APPROVED** [review]

**Pull request merged**